File: dotnet/Pages/ProductPage.cs
using System.Threading.Tasks;
using Microsoft.Playwright;

namespace PlaywrightCapstone.Pages
{
  /// <summary>
  /// Page Object Model for product pages. Keep selectors centralized here so
  /// if the demo app markup changes, only this class needs updates.
  /// </summary>
  public class ProductPage
  {
    private readonly IPage _page;

    public ProductPage(IPage page)
    {
      _page = page;
    }

    /// <summary>
    /// Click a product by visible name text. Example: "Laptop".
    /// </summary>
    public async Task SelectProductByNameAsync(string name)
    {
      await _page.ClickAsync($"text={name}");
    }

    /// <summary>
    /// Add the currently selected product to cart using a standard id-based selector.
    /// </summary>
    public async Task AddToCartAsync()
    {
      await _page.ClickAsync("#add-to-cart");
    }
  }
}

File: java/src/test/java/pages/ProductPage.java
package pages;

import com.microsoft.playwright.Page;

/**
 * POM for Product page.
 * Provides simple helpers to select a product by its visible name and add it to
 * cart.
 */
public class ProductPage {
  private final Page page;

  public ProductPage(Page page) {
    this.page = page;
  }

  /**
   * Select product by its visible text.
   *
   * @param name Product name to click
   */
  public void selectProductByName(String name) {
    page.click("text=" + name);
  }

  public void addToCart() {
    page.click("#add-to-cart");
  }
}
